src/parse.rs

```diff
@@ -364,6 +364,17 @@ impl<'a> Bytes<'a> {
     where
         T: FromStr,
     {
+        if self.consume_ident("inf") {
```

Comment:
sounds like these conditions could be simplified (with a loop or something)

Reply:
Done